pkgs/stdenv/generic/check-meta.nix

```diff
@@ -80,14 +80,41 @@ let
   pos_str = meta: meta.position or "«unknown-file»";
 
   remediation = {
-    unfree = remediate_whitelist "Unfree";
+    unfree = remediate_unfree;
     broken = remediate_whitelist "Broken";
     unsupported = remediate_whitelist "UnsupportedSystem";
     blacklisted = x: "";
     insecure = remediate_insecure;
     broken-outputs = remediateOutputsToInstall;
     unknown-meta = x: "";
   };
+  remediate_unfree = attrs:
+    ''
+      a) For `nixos-rebuild` you can allow unfree packages globally:
+        { nixpkgs.config.allowUnfree = true; }
+
+      Or configure a predicate to whitelist specific packages:
+        { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
+            "package-name-to-whitelist"
+          ];
+        }
+
+      in configuration.nix to override this.
+
+      b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
+        { allowUnfree = true; }
```

Comment:
or by setting the environment-variable `NIXPKGS_ALLOW_UNFREE=1`, which especially useful when unfree packages are required only occasionally with `nix search`, `nix run` or `nix-shell`.